react-form-controls/src/NewsletterForm.jsx

```diff
@@ -0,0 +1,39 @@
+import React from "react"
+
+class NewsletterForm extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {};
```

Comment:
Your state object should have starting values for all values in state you are going to use